apps/meteor/app/livechat/server/api/v1/room.js

```diff
@@ -199,10 +199,20 @@ API.v1.addRoute('livechat/room.survey', {
 
 API.v1.addRoute(
 	'livechat/room.forward',
-	{ authRequired: true },
+	{ authRequired: true, permissionsRequired: ['view-l-room', 'transfer-livechat-guest'] },
 	{
 		post() {
-			API.v1.success(Meteor.runAsUser(this.userId, () => Meteor.call('livechat:transfer', this.bodyParams)));
+			let result = false;
+
+			Meteor.runAsUser(this.userId, () => {
+				result = Meteor.call('livechat:transfer', this.bodyParams);
+			});
+
+			if (result) {
+				return API.v1.success();
+			}
+
+			return API.v1.failure();
```

Comment:
Is the UI prepared for this change? Otherwise, this would be a breaking change

Reply:
Actually, this endpoint wasn't getting used on the UI before - we were still using a meteor method over there. Hence I've also made the transition on UI to use this endpoint instead of the method